stl/inc/ranges

```diff
@@ -6837,6 +6837,602 @@ namespace ranges {
+            _Modified_iterator -= _Rhs;
+
+            return _Modified_iterator;
+        }
+
+        _NODISCARD_FRIEND constexpr difference_type operator-(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+            noexcept(_Zip::_Get_smallest_iterator_sentinel_distance<difference_type>(
+                _Lhs._M_Current, _Rhs._M_Current))) // strengthened
+            requires(sized_sentinel_for<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>,
+                iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>&&...) {
+            return _Zip::_Get_smallest_iterator_sentinel_distance<difference_type>(_Lhs._M_Current, _Rhs._M_Current);
+        }
+
+        _NODISCARD_FRIEND constexpr auto iter_move(const _Iterator& _Itr) noexcept(
+            (noexcept(_RANGES iter_move(_STD declval<const iterator_t<_Maybe_const<_IsConst, _ViewTypes>>&>())) && ...)
+            && (is_nothrow_move_constructible_v<range_rvalue_reference_t<_Maybe_const<_IsConst, _ViewTypes>>> && ...)) {
+            return _Zip::_Tuple_transform(_RANGES iter_move, _Itr._M_Current);
+        }
+
+        friend constexpr void iter_swap(const _Iterator& _Lhs, const _Iterator& _Rhs) noexcept(
+            (noexcept(iter_swap(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>(),
+                 _STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>()))
+                && ...)) requires(indirectly_swappable<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>&&...) {
+            const auto _Swap_every_pair_lambda = [&_Lhs, &_Rhs ]<size_t... _Indices>(index_sequence<_Indices...>) {
+                ((_RANGES iter_swap(_STD get<_Indices>(_Lhs._M_Current), _STD get<_Indices>(_Rhs._M_Current))), ...);
+            };
+
+            constexpr auto _Index_sequence = index_sequence_for<_ViewTypes...>{};
+            _Swap_every_pair_lambda(_Index_sequence);
+        }
+
+    private:
+        _Iterator_tuple_type _M_Current;
+    };
+
+    template <input_range... _ViewTypes>
+        requires _Zip::_Allow_views<_ViewTypes...>
+    template <bool _IsConst>
+    class zip_view<_ViewTypes...>::_Sentinel {
+    private:
+        friend class zip_view<_ViewTypes...>;
+
+        using _Sentinel_tuple_type = tuple<sentinel_t<_Maybe_const<_IsConst, _ViewTypes>>...>;
+
+        constexpr explicit _Sentinel(_Sentinel_tuple_type _End) noexcept(
+            is_nothrow_copy_assignable_v<_Sentinel_tuple_type>) // strengthened
+            : _M_End(_End) {}
+
+        template <bool _IteratorConst>
+        using _Iterator_sentinel_difference_type =
+            common_type_t<range_difference_t<_Maybe_const<_IteratorConst, _ViewTypes>>...>;
+
+        template <bool _IteratorConst>
+        static constexpr bool _Iterator_sentinel_equality_noexcept = noexcept(
+            _Zip::_Evaluate_iterator_sentinel_equality(
+                _Get_iterator_tuple(
+                    _STD declval<const typename zip_view<_ViewTypes...>::template _Iterator<_IteratorConst>&>()),

```

Comment:
`const typename zip_view<_ViewTypes...>::template _Iterator<_IteratorConst>&` occurs 5 times within `_Sentinel`. For clarity, could this be extracted into an alias template, so `_Sentinel` could just say `const SOMETHING<_IteratorConst>&`? (It's probably best to keep the `const` and `&` "outside" the alias, again for clarity.)